aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/AddVersionCommand.java

```diff
@@ -152,19 +154,21 @@ public AddVersionCommand(RestApiService restApiService, JobsService jobsService,
     @Override
     public Integer call() {
         ApiInfoDto apiInfo = null;
+        DebugOptionsService debugOptionsService;
         try {
             if (sharedOptions.getTimeout() != Constants.DEFAULT_HTTP_TIMEOUT) {
                 restApiService.setTimeout(sharedOptions.getTimeout(), TimeUnit.SECONDS);
             }
             restApiService.validateUrlAndKey(sharedOptions.getFullServerRootUrl(), sharedOptions.getUsername(), sharedOptions.getApiKeyValue());
             apiInfo = restApiService.getAipConsoleApiInfo();
+            debugOptionsService = new DebugOptionsServiceImpl(restApiService, apiInfo);
```

Comment:
???? why recreate instance of a bean

Reply:
This is not a bean in sens that is neither @Componen nor @Service.
It is just a typical class to manage debug options compatibility.. Why not choosing a bean? It is just because at time it will be created the apiInfo will not be available.

Reply:
it is a bean, you even injected the other services to that.